### PR TITLE
feat: upgrade to Patternfly v5

### PR DESCRIPTION
**Pull request by mmelko**
I used `@patternfly/pf-codemods` to upgrade most of the things, then fixed errors and some warnings. And upgraded some tables. In the next iteration I will ged rid of the rest deprecated components, mainly Selects and Dropdowns. 

`branch: jsolovjo:e2e-fix-patternfly5-tests`

### CODE REVIEW

**Review by grgrzybek — APPROVED** [review]

**Review by tadayosi — COMMENTED** [review]
the changes too big. first review batch.

File: packages/hawtio/src/plugins/connect/ConnectionStatus.tsx [hunk 1]
```diff
@@ -35,13 +36,25 @@ export const ConnectionStatus: React.FunctionComponent = () => {
   let icon = null
   switch (reachable) {
     case 'reachable':
-      icon = <PluggedIcon color='green' />
+      icon = (
+        <Icon style={{ color: 'var(--pf-v5-global--success-color--100)' }}>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/connect/ConnectionStatus.tsx:40]: why can't we use `color={'xxx'}` here?

File: packages/hawtio/src/plugins/connect/ConnectionStatus.tsx [hunk 2]
```diff
@@ -35,13 +36,25 @@ export const ConnectionStatus: React.FunctionComponent = () => {
...
+          <PluggedIcon />
+        </Icon>
+      )
       break
     case 'not-reachable':
-      icon = <UnpluggedIcon color='red' />
+      icon = (
+        <Icon style={{ color: 'var(--pf-v5-global--danger-color--100)' }}>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/connect/ConnectionStatus.tsx:47]: why can't we use `color={'xxx'}` here?

File: packages/hawtio/src/plugins/connect/ConnectionStatus.tsx [hunk 3]
```diff
@@ -35,13 +36,25 @@ export const ConnectionStatus: React.FunctionComponent = () => {
...
+          <UnpluggedIcon />
+        </Icon>
+      )
       break
     case 'not-reachable-securely':
-      icon = <PluggedIcon style={{ color: 'var(--pf-global--warning-color--100)' }} />
+      icon = (
+        <Icon style={{ color: 'var(--pf-v5-global--warning-color--100)' }}>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/connect/ConnectionStatus.tsx:54]: why can't we use `color={'xxx'}` here?

**Comment by grgrzybek** [comment]
Thanks @tadayosi for careful review - I admit, I simply assumed Matej did the best.

File: packages/hawtio/src/plugins/shared/operations/OperationForm.tsx [hunk 4]
```diff
@@ -317,7 +322,7 @@ const ArgFormInput: React.FunctionComponent<{
           id={id}
           type='number'
           value={Number(value)}
-          onChange={updateArgValues(index)}
+          onChange={() => updateArgValues(index)}
```
[the remark this answers is not in the document]
Comment (tadayosi) [reply packages/hawtio/src/plugins/shared/operations/OperationForm.tsx:325]: ok, if you are doing this just to satisfy the compiler you are doing something wrong indeed. for curried functions, it actually shouldn't do anything other than returning the function, so it should break the code. really need to check similar changes again.

File: packages/hawtio/src/ui/icons/TooltipHelpIcon.tsx [hunk 5]
```diff
@@ -3,7 +3,7 @@ import { HelpIcon } from '@patternfly/react-icons'
 
 export const TooltipHelpIcon: React.FunctionComponent<{ tooltip: string }> = ({ tooltip }) => (
   <Icon size='md'>
-    <Tooltip content={tooltip} removeFindDomNode>
+    <Tooltip content={tooltip}>
```
Comment (tadayosi) [inline packages/hawtio/src/ui/icons/TooltipHelpIcon.tsx:6]: is it intentional? no more needed?

File: packages/hawtio/src/ui/login/HawtioLogin.tsx [hunk 6]
```diff
@@ -58,7 +58,7 @@ export const HawtioLogin: React.FunctionComponent = () => {
 
   return (
     <LoginPage
-      backgroundImgSrc={backgroundImages}
+      backgroundImgSrc={backgroundImages.lg as string}
```
Comment (tadayosi) [inline packages/hawtio/src/ui/login/HawtioLogin.tsx:61]: I don't like such adhoc type casting. do we really need this or shouldn't we change the code in a different way?

File: packages/hawtio/src/ui/page/HawtioPage.tsx [hunk 7]
```diff
@@ -51,7 +58,7 @@ export const HawtioPage: React.FunctionComponent = () => {
 
   return (
     <PageContext.Provider value={{ username, plugins }}>
-      <BackgroundImage src={backgroundImages} />
+      <BackgroundImage src={backgroundImages.lg} />
```
Comment (tadayosi) [inline packages/hawtio/src/ui/page/HawtioPage.tsx:61]: why the change?

**Review by tadayosi — CHANGES_REQUESTED** [review]
finished review. at this point I'd like to request changes, as the way you changed around the curried callback functions is alarming and suspected that it breaks the code. please check them again.

File: packages/hawtio/src/plugins/connect/ConnectionStatus.tsx (hunk 2 above)
Comment (grgrzybek) [reply packages/hawtio/src/plugins/connect/ConnectionStatus.tsx:47]: that's my change. I didn't find proper color and I was using it from https://www.patternfly.org/components/alert/#prefixed-with-pf-v5-c-alert

File: packages/hawtio/src/plugins/connect/ConnectionStatus.tsx (hunk 3 above)
Comment (grgrzybek) [reply packages/hawtio/src/plugins/connect/ConnectionStatus.tsx:54]: same here

File: packages/hawtio/src/core/event-service.ts [hunk 8]
```diff
@@ -3,7 +3,7 @@ import { Logger } from './logging'
 
 const log = Logger.get('hawtio-core-event')
 
-export type NotificationType = 'default' | 'info' | 'success' | 'warning' | 'danger'
+export type NotificationType = 'info' | 'success' | 'warning' | 'danger' | 'custom'
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/src/core/event-service.ts:6]: yes for the notification.

File: packages/hawtio/package.json [hunk 9]
```diff
@@ -36,12 +36,12 @@
...
-    "@patternfly/react-core": "~4.278.1",
-    "@patternfly/react-table": "~4.113.7",
-    "@testing-library/jest-dom": "^6.4.5",
-    "@testing-library/react": "^15.0.7",
+    "@patternfly/react-charts": "~7.3.0",
+    "@patternfly/react-code-editor": "~5.3.3",
+    "@patternfly/react-core": "~5.3.3",
+    "@patternfly/react-table": "~5.3.3",
+    "@testing-library/jest-dom": "^6.4.2",
+    "@testing-library/react": "^15.0.6",
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/package.json:44]: thanks for the catch, I probably did it accidentally when resolving the conflicts.

File: packages/hawtio/src/plugins/camel/CamelPreferences.tsx [hunk 10]
```diff
@@ -39,7 +39,7 @@ export const CamelPreferences: React.FunctionComponent = () => {
               id='camel-form-maximum-label-width-input'
               type='number'
               value={options.maximumLabelWidth}
-              onChange={updateNumberValueFor('maximumLabelWidth')}
+              onChange={() => updateNumberValueFor('maximumLabelWidth')}
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/src/plugins/camel/CamelPreferences.tsx:42]: I will look at this, the true is that there is new event parameter, and compiler stopped complaining. I totally missed that it's a curried function. Same with others. I will refactor and update.

File: packages/hawtio/src/plugins/shared/operations/OperationForm.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/src/plugins/shared/operations/OperationForm.tsx:325]: yeah, you're right as I mentioned above.

File: packages/hawtio/src/plugins/connect/ConnectionStatus.tsx (hunk 2 above)
Comment (mmelko) [reply packages/hawtio/src/plugins/connect/ConnectionStatus.tsx:47]: I updated the rest icons with same as Grzegorz because I had all of them black.

File: packages/hawtio/src/ui/login/HawtioLogin.tsx (hunk 6 above)
Comment (mmelko) [reply packages/hawtio/src/ui/login/HawtioLogin.tsx:61]: Image sources now required strings not BackgroundImageSrcMap's

File: packages/hawtio/src/ui/page/HawtioPage.tsx (hunk 7 above)
Comment (mmelko) [reply packages/hawtio/src/ui/page/HawtioPage.tsx:61]: Image sources now required strings not BackgroundImageSrcMap's

**Comment by mmelko** [comment]
@tadayosi thanks for the review I will comment. 

> Thanks @tadayosi for careful review - I admit, I simply assumed Matej did the best.

Well I did:), or at least I tried diven that time and space ;)

**Comment by mmelko** [comment]
> finished review. at this point I'd like to request changes, as the way you changed around the curried callback functions is alarming and suspected that it breaks the code. please check them again.

Thanks for the review, as I mentioned you're right, I missed curried signatures of the functions. It also discovered other thing, and that is that are unit test suite could be better too.

File: packages/hawtio/src/plugins/camel/debug/Debug.tsx [hunk 11]
```diff
@@ -25,7 +25,7 @@ import {
   PlusIcon,
   TimesCircleIcon,
 } from '@patternfly/react-icons'
-import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table'
+import { Table /* data-codemods */, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table'
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/src/plugins/camel/debug/Debug.tsx:28]: that's comment automatically added by the codemods that it was modified by the tool

File: packages/hawtio/src/plugins/shared/operations/OperationForm.tsx [hunk 12]
```diff
@@ -163,7 +163,6 @@ const OperationFormContents: React.FunctionComponent<{ isExpanded: boolean }> =
         isExpanded
         isCode
         isReadOnly
-        removeFindDomNode
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/src/plugins/shared/operations/OperationForm.tsx:166]: yes, it was removed by codemods tool. It's now the default behaviour see: https://github.com/patternfly/patternfly-react/pull/8371

File: packages/hawtio/src/ui/icons/TooltipHelpIcon.tsx (hunk 5 above)
Comment (mmelko) [reply packages/hawtio/src/ui/icons/TooltipHelpIcon.tsx:6]: yes, as I mentioned above, it's not longer needed

File: packages/hawtio/src/plugins/shared/chart/WatchableAttributesForm.tsx [hunk 13]
```diff
@@ -86,7 +87,11 @@ export const WatchableAttributesForm = ({
 
             onAttributesToWatchUpdate(newWatches)
             setModalAttributesToWatch({ ...newWatches })
-          }) as unknown as (newAvailableOptions: ReactNode[], newChosenOptions: ReactNode[]) => void
+          }) as unknown as (
+            _event: React.MouseEvent<HTMLElement, MouseEvent>,
+            newAvailableOptions: ReactNode[],
+            newChosenOptions: ReactNode[],
+          ) => void
```
[the remark this answers is not in the document]
Comment (mmelko) [reply packages/hawtio/src/plugins/shared/chart/WatchableAttributesForm.tsx:96]: well honestly, I'd rather rework the form in the future.

File: packages/hawtio/src/plugins/shared/chart/Chart.tsx [hunk 14]
```diff
@@ -279,11 +273,13 @@ export const Chart: React.FunctionComponent = () => {
...
-          <CardHeader>
-            <CardActions>
-              <Button onClick={() => setIsWatchableAttributesModalOpen(true)}>Edit watches</Button>
-            </CardActions>
-          </CardHeader>
+          <CardHeader
+            actions={{
+              actions: <Button onClick={() => setIsWatchableAttributesModalOpen(true)}>Edit watches</Button>,
+              hasNoOffset: false,
+              className: undefined,
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/shared/chart/Chart.tsx:280]: maybe it's not necessary then

File: packages/hawtio/src/plugins/shared/chart/Chart.tsx [hunk 15]
```diff
@@ -307,16 +303,19 @@ export const Chart: React.FunctionComponent = () => {
       <Grid hasGutter span={12} xl2={6}>
         <GridItem span={12}>
           <Card>
-            <CardHeader>
+            <CardHeader
+              actions={{
+                actions: <Button onClick={() => setIsWatchableAttributesModalOpen(true)}>Edit watches</Button>,
+                hasNoOffset: false,
+                className: undefined,
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/shared/chart/Chart.tsx:310]: maybe it's not necessary then

File: packages/hawtio/src/plugins/springboot/Health.tsx [hunk 16]
```diff
@@ -32,21 +32,41 @@ const ComponentDetails: React.FunctionComponent<{ componentDetails: HealthCompon
...
-    </TableComposable>
+    </Table>
   )
 }
 const HealthStatusIcon: React.FunctionComponent<{ status: string }> = ({ status }) => {
   switch (status) {
     case 'UP':
-      return <CheckCircleIcon color={'green'} />
+      return (
+        <Icon color={'green'}>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/springboot/Health.tsx:42]: If your comment in another place is right, shouldn't then this be changed to use the css style directly?

File: packages/hawtio/src/plugins/springboot/Health.tsx [hunk 17]
```diff
@@ -32,21 +32,41 @@ const ComponentDetails: React.FunctionComponent<{ componentDetails: HealthCompon
...
+          <CheckCircleIcon />
+        </Icon>
+      )
     case 'DOWN':
-      return <ExclamationCircleIcon color={'red'} />
+      return (
+        <Icon color={'red'}>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/springboot/Health.tsx:48]: same here

File: packages/hawtio/src/plugins/springboot/Health.tsx [hunk 18]
```diff
@@ -32,21 +32,41 @@ const ComponentDetails: React.FunctionComponent<{ componentDetails: HealthCompon
...
+          <ExclamationCircleIcon />
+        </Icon>
+      )
     case 'OUT_OF_SERVICE':
-      return <ExclamationTriangleIcon color={'orange'} />
+      return (
+        <Icon color={'orange'}>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/springboot/Health.tsx:54]: same here

File: packages/hawtio/src/plugins/springboot/Health.tsx [hunk 19]
```diff
@@ -32,21 +32,41 @@ const ComponentDetails: React.FunctionComponent<{ componentDetails: HealthCompon
...
-      return <QuestionCircleIcon />
+      return (
+        <Icon>
+          <QuestionCircleIcon />
+        </Icon>
+      )
     default:
-      return <InfoCircleIcon color={'blue'} />
+      return (
+        <Icon color={'blue'}>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/springboot/Health.tsx:66]: same here

File: packages/hawtio/src/plugins/springboot/TraceView.tsx [hunk 20]
```diff
@@ -22,14 +19,22 @@ import {
...
+import { Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table'
 import { CheckCircleIcon, ExclamationCircleIcon, SearchIcon } from '@patternfly/react-icons'
 import { Trace } from './types'
 import { springbootService } from './springboot-service'
 
 const HttpStatusIcon: React.FunctionComponent<{ code: number }> = ({ code }) => {
-  if (code < 400) return <CheckCircleIcon color='#3E8635' />
+  if (code < 400)
+    return (
+      <Icon color='#3E8635'>
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/springboot/TraceView.tsx:34]: same here

File: packages/hawtio/src/plugins/springboot/TraceView.tsx [hunk 21]
```diff
@@ -22,14 +19,22 @@ import {
...
+        <CheckCircleIcon />
+      </Icon>
+    )
   else return <ExclamationCircleIcon color='#C9190B' />
```
Comment (tadayosi) [inline packages/hawtio/src/plugins/springboot/TraceView.tsx:38]: same here

File: packages/hawtio/src/ui/login/HawtioLogin.tsx (hunk 6 above)
Comment (tadayosi) [reply packages/hawtio/src/ui/login/HawtioLogin.tsx:61]: The point is that we shouldn't use `as string` in an adhoc way. Why don't you change the definition of `backgroundImages` then?
https://github.com/hawtio/hawtio-next/blob/main/packages/hawtio/src/img/index.ts

Similarly, probably we should change the background images we use to conform to those of PF 5. The images currently used just come from PF 4 assets.
https://www.patternfly.org/components/login-page

File: packages/hawtio/src/ui/page/HawtioPage.tsx (hunk 7 above)
Comment (tadayosi) [reply packages/hawtio/src/ui/page/HawtioPage.tsx:61]: Then the problem should be that now the background image doesn't change automatically depending on the size of the display. Your change seems to force using only the lg size image for every display.

How can we adjust background image size based on the display resolution with PF 5?

**Review by tadayosi — COMMENTED** [review]
Resolved some and added/updated some more comments

File: packages/hawtio/src/ui/login/HawtioLogin.tsx (hunk 6 above)
Comment (tadayosi) [reply packages/hawtio/src/ui/login/HawtioLogin.tsx]: we'll work on the issue at another pull req.

**Review by tadayosi — APPROVED** [review]
